lib/tools/apk-utils.js

```diff
@@ -41,6 +41,8 @@ apkUtilsMethods.startApp = async function (startAppOptions = {}) {
       log.errorAndThrow("activity and pkg is required for launching application");
     }
     startAppOptions = _.clone(startAppOptions);
+    startAppOptions.activity = startAppOptions.activity.replace('$', '\\$');
```

Comment:
It's covered on l40, right? So it'll throw and never get this far if we don't have an activity defined.